src/modules/avformat/consumer_avformat.c

```diff
@@ -1851,6 +1851,15 @@ static void *consumer_thread( void *arg )
 						converted_avframe->quality = c->global_quality;
 						converted_avframe->pts = frame_count;
 
+						// set frame interlaced if required but still progressive
+						if ( mlt_properties_get_int( frame_properties, "progressive" ) &&
+							mlt_properties_get_int( frame_properties, "consumer_tff" ) )
```

Comment:
I faced strange situation then source frame is **progressive** but consumer want **interlaced**. None of processing switch it back to **interlaced** and as result codec accept progressive frame instead of **interlaced**

Reply:
This is what I thought. I do not know if it is bad to give progressive frames to codecs configured for interlaced: ildct+ilme. I think it might be, and if so this is a nice fix to have.

Reply:
Replace `mlt_properties_get_int( frame_properties, "consumer_tff" )` with `!mlt_properties_get_int( consumer, "progressive" )`
https://github.com/mltframework/mlt/blob/master/src/framework/mlt_consumer.h#L67